### PR TITLE
Tests new API

### PR DESCRIPTION
**Pull request by elad-c**

### CODE REVIEW

File: tests/keras_tests/feature_networks_tests/feature_networks/old_api_test.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                                                     n_iter=self.num_calibration_iter,
+                                                     quant_config=quant_config,
+                                                     gptq_config=gptq_config,
+                                                     target_platform_capabilities=self.get_tpc(),
+                                                     )
+
+        self.compare(ptq_model, model_float, input_x=self.representative_data_gen(),
+                     quantization_info=quantization_info)
+
+    def compare(self, quant_model, float_model, input_x=None, quantization_info=None):
```
Comment (reuvenperetz) [inline tests/keras_tests/feature_networks_tests/feature_networks/old_api_test.py:81]: Maybe this test shouldn't check a thing (since it's only for coverage purposes)?

File: tests/keras_tests/feature_networks_tests/feature_networks/old_api_test.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+from model_compression_toolkit.core.tpc_models.default_tpc.latest import get_op_quantization_configs
+
+
+keras = tf.keras
+layers = keras.layers
+
+
+class OldApiTest(BaseKerasFeatureNetworkTest):
+    def __init__(self, unit_test, mp_enable=False, gptq_enable=False):
+        super().__init__(unit_test, val_batch_size=1, num_calibration_iter=100)
```
Comment (reuvenperetz) [inline tests/keras_tests/feature_networks_tests/feature_networks/old_api_test.py:30]: Why do we need 100 iterations?

File: tests/keras_tests/feature_networks_tests/feature_networks/second_moment_correction_test.py [hunk 3]
```diff
@@ -327,13 +318,11 @@ def run_test(self):
...
-                                                                                  self.representative_data_gen,
-                                                                                  core_config=core_cfg,
-                                                                                  target_platform_capabilities=self.get_tpc(),
-                                                                                  new_experimental_exporter=True)
-
+                ptq_model, quantization_info = self.get_ptq_facade()(model_float,
+                                                                     self.representative_data_gen_experimental,
+                                                                     core_config=self.get_core_config(),
+                                                                     target_platform_capabilities=self.get_tpc(),
+                                                                     new_experimental_exporter=True)
```
Comment (reuvenperetz) [inline tests/keras_tests/feature_networks_tests/feature_networks/second_moment_correction_test.py:325]: Use self.experimental_exporter instead of True

File: tests/keras_tests/feature_networks_tests/test_features_runner.py [hunk 4]
```diff
@@ -546,10 +546,6 @@ def test_gptq(self):
                         rounding_type=RoundingType.SoftQuantizer,
                         quantizer_config=SoftQuantizerConfig(),
                         per_channel=True).run_test()
-        GradientPTQTest(self,
-                        rounding_type=RoundingType.SoftQuantizer,
-                        quantizer_config=SoftQuantizerConfig(),
-                        per_channel=True).run_test(experimental_facade=True)
```
Comment (reuvenperetz) [inline tests/keras_tests/feature_networks_tests/test_features_runner.py:552]: Why this test has been removed?

File: tests/pytorch_tests/model_tests/feature_models/old_api_test.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    def forward(self, x):
+        x = self.conv1(x)
+        x = self.relu(x)
+        return x
+
+
+class OldApiTest(BasePytorchTest):
+    def __init__(self, unit_test, mp_enable=False, gptq_enable=False):
+        super().__init__(unit_test)
+        self.num_calibration_iter = 100
```
Comment (reuvenperetz) [inline tests/pytorch_tests/model_tests/feature_models/old_api_test.py:42]: Why 100?

File: tests/pytorch_tests/model_tests/feature_models/old_api_test.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+                                                     n_iter=self.num_calibration_iter,
+                                                     quant_config=quant_config,
+                                                     gptq_config=gptq_config,
+                                                     target_platform_capabilities=self.get_tpc()['all_32bit'],
+                                                     )
+
+        self.compare(ptq_model, model_float, input_x=self.representative_data_gen(),
+                     quantization_info=quantization_info)
+
+    def compare(self, quant_model, float_model, input_x=None, quantization_info=None):
```
Comment (reuvenperetz) [inline tests/pytorch_tests/model_tests/feature_models/old_api_test.py:117]: Same as in TF, maybe just pass?

File: tests/keras_tests/feature_networks_tests/feature_networks/old_api_test.py (hunk 1 above)
Comment (elad-c) [reply tests/keras_tests/feature_networks_tests/feature_networks/old_api_test.py:81]: I think it's necessary to verify the returned quantized model is valid. The facade itself might introduce bugs that don't result in failed runs, but failed accuracy.

File: tests/keras_tests/feature_networks_tests/feature_networks/old_api_test.py (hunk 2 above)
Comment (elad-c) [reply tests/keras_tests/feature_networks_tests/feature_networks/old_api_test.py:30]: For better accuracy in the Compare method

File: tests/keras_tests/feature_networks_tests/test_features_runner.py (hunk 4 above)
Comment (elad-c) [reply tests/keras_tests/feature_networks_tests/test_features_runner.py:552]: it's a duplicate test. Enough to run without the experimantal_facase parameter, as itno longer exists

File: tests/pytorch_tests/model_tests/feature_models/old_api_test.py (hunk 5 above)
Comment (elad-c) [reply tests/pytorch_tests/model_tests/feature_models/old_api_test.py:42]: Accuracy...

**Review by reuvenperetz — APPROVED** [review]
